test/libsolidity/ASTJSONTest.cpp

```diff
@@ -105,6 +116,15 @@ ASTJSONTest::ASTJSONTest(string const& _filename)
 			);
 			source = string();
 		}
+		else if (boost::algorithm::starts_with(line, failMarker))
+		{
+			string state = line.substr(failMarker.size(), line.size());
+			state.erase(std::remove_if(state.begin(), state.end(),
+									   [](unsigned char x){return std::isspace(x);}),
+						state.end());
```

Comment:
Some wonky indentation here. Also a bit hard to see which argument goes into which function and which opening parentheses match which closing ones.

This is the style we typically use:
```suggestion
			state.erase(
				remove_if(
					state.begin(),
					state.end(),
					[](unsigned char x){ return isspace(x); }
				),
				state.end()
			);
```

Reply:
Also, you can just use `boost::algorithm::string::trim()` here. I think your version removes all spaces (not just trailing ones) but I don't think it's a good idea. I think that trimming leading and trailing whitespace is enough.